src/ooxml-file-cache.ts

```diff
@@ -289,13 +290,15 @@ export class OOXMLFileCache {
       await workspace.fs.createDirectory(Uri.file(dirname(cachedFilePath)));
       await workspace.fs.writeFile(Uri.file(cachedFilePath), fileContents);
     } catch (err) {
-      const errorType = throwErrorForType ? throwErrorForType.toLowerCase() : '';
-      if (errorType && (err?.code.toLowerCase() === errorType || err?.message.toLowerCase().includes(errorType))) {
-        throw new Error(err);
+      const e = (err as FileSystemError);
+
+      if (
```

Comment:
We don't want to expose all exceptions only EBUSY. The EBUSY means the file is open in another program and they should close it before updating the xml parts. Other errors should be handled normally.

Reply:
We can still check for EBUSY but I think it'd be better if we did that check in `updateOOXMLFile()` in ooxml-viewer.ts, instead of this method.  Based on testing, it looks like when the file opened in another program, the it throws a `FileSystemError` where `err.code` is always `Unknown` and `err.message` contains `EBUSY`.  I think we should be looking for that specific condition instead or a just looking for EBUSY in both `err.code` and `err.message` and that we should do that in `updateOOXMLFile()`

Reply:
I moved the error checking to updateOOXMLFile.